### PR TITLE
request work_available from bouncer.

### PR DESCRIPTION
**Pull request by nxvl**
@rainforestapp/tester-product @rainforestapp/core @ukd1

### CODE REVIEW

**Comment by ukd1** [comment]
@nxvl lgtm, but please bump version id too

**Comment by mojenLXXXIII** [comment]
@nxvl  what dos it mean to get job from BOUNCER_URL?

**Comment by ElPeter** [comment]
@mojenLXXXIII 

I think they are updating the extension to the guaranteed assignments, so no more "Ooops!" pages...

Are you guys getting things done today cause I'm waiting for some work and I haven't received a hit since you bumped it...

**Comment by nxvl** [comment]
Yes we are actually working on avoiding those oops and for it to be faster and more reliable.

**Comment by mojenLXXXIII** [comment]
@ElPeter most of RF updates take place late Friday hope they update this one early

File: extension/manifest.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "Rainforest Tester Notifier",
-  "version": "1.0.0",
+  "version": "1.0.1",
```
Comment (usami) [inline extension/manifest.json]: nitpicking, but maybe minor version change? (so `1.1.0`?)

File: extension/js/main.js [hunk 2]
```diff
@@ -30,6 +34,20 @@ chrome.storage.sync.get("worker_uuid", function(data) {
   }
 });
 
+//
+// Load the initial api endpoint value from storage
+//
+chrome.storage.sync.get("work_available_endpoint", function(data) {
+  // Notify that we saved.
+  console.log("Data loaded", data);
```
Comment (juliankrispel) [inline extension/js/main.js]: remove the console.log

File: extension/js/main.js [hunk 3]
```diff
@@ -30,6 +34,20 @@ chrome.storage.sync.get("worker_uuid", function(data) {
...
+  if (data['work_available_endpoint'] != undefined) {
+    info_hash["work_available_endpoint"] = data["work_available_endpoint"];
+    console.log("Updated info hash:", info_hash);
```
Comment (juliankrispel) [inline extension/js/main.js]: again

File: extension/js/main.js [hunk 4]
```diff
@@ -30,6 +34,20 @@ chrome.storage.sync.get("worker_uuid", function(data) {
   }
 });
 
+//
+// Load the initial api endpoint value from storage
+//
+chrome.storage.sync.get("work_available_endpoint", function(data) {
+  // Notify that we saved.
+  console.log("Data loaded", data);
+  if (data['work_available_endpoint'] != undefined) {
```
Comment (juliankrispel) [inline extension/js/main.js]: always use strict comparison operators in this case `!==`

File: extension/js/main.js [hunk 5]
```diff
@@ -41,7 +59,7 @@ chrome.browserAction.onClicked.addListener(function (event) {
   console.log("browserAction", event);
   console.log("Checking is now:", _checking_active);
```
Comment (juliankrispel) [inline extension/js/main.js]: remove console logs

File: extension/js/main.js [hunk 6]
```diff
@@ -52,15 +70,22 @@ chrome.runtime.onMessageExternal.addListener(function(request, sender, sendRespo
   if (request.data) {
     console.log("incoming data", request);
```
Comment (juliankrispel) [inline extension/js/main.js]: remove

File: extension/js/main.js [hunk 7]
```diff
@@ -52,15 +70,22 @@ chrome.runtime.onMessageExternal.addListener(function(request, sender, sendRespo
...
 
-    if (request.data['worker_uuid']) {
-      info_hash["uuid"] = request.data["worker_uuid"]
+    if (request.data['worker_uuid'] && request.data['work_available_endpoint']) {
+      info_hash["uuid"] = request.data["worker_uuid"];
```
Comment (juliankrispel) [inline extension/js/main.js]: you can and should write this as request.data.worker_uuid

File: extension/js/main.js [hunk 8]
```diff
@@ -52,15 +70,22 @@ chrome.runtime.onMessageExternal.addListener(function(request, sender, sendRespo
...
       set_checking(_checking_active);
       sendResponse({ok: true});
 
-      chrome.storage.sync.set({'worker_uuid': request.data["worker_uuid"]}, function() {
-        // Notify that we saved.
-        console.log('Worker ID saved');
-      });
+      chrome.storage.sync.set(
+        {
+          'worker_uuid': request.data["worker_uuid"],
```
Comment (juliankrispel) [inline extension/js/main.js]: keys shouldn't be strings

File: extension/js/main.js [hunk 9]
```diff
@@ -52,15 +70,22 @@ chrome.runtime.onMessageExternal.addListener(function(request, sender, sendRespo
...
+          'work_available_endpoint': request.data["work_available_endpoint"]
+        },
+        function() {
+          // Notify that we saved.
+          console.log('Worker ID and API endpoint saved');
```
Comment (juliankrispel) [inline extension/js/main.js]: remove

File: extension/js/main.js [hunk 10]
```diff
@@ -5,16 +5,20 @@ console.log("Starting:", manifest.name, manifest.version, BASE_URL);
 
 // Start disabled: require the tester to enable if they want to
 // work when the browser starts
-_checking_active = false
-info_hash = {tester_state: 'active', email: '', id: '', version: manifest.version}
+_checking_active = false;
+info_hash = {
```
Comment (juliankrispel) [inline extension/js/main.js]: these variables aren't declared, use `var` to declare them

**Comment by juliankrispel** [comment]
this whole app is a javascript crime :heart:, I'd suggest you install `jshint` and run autocorrect on it, other then than and all the console.logs, looks good to me :)

File: extension/js/main.js [hunk 11]
```diff
@@ -143,12 +159,12 @@ function sync_tab_make_new() {
 //
 function check_for_work() {
   if (info_hash["uuid"] == "" || info_hash["uuid"] === undefined) {
-    console.log("Info hash not set", info_hash)
+    console.log("Info hash not set", info_hash);
```
Comment (juliankrispel) [inline extension/js/main.js]: please remove all console logs

Comment (juliankrispel) [inline extension/js/main.js]: that's the last one actually

File: extension/js/main.js [hunk 12]
```diff
@@ -30,6 +34,18 @@ chrome.storage.sync.get("worker_uuid", function(data) {
   }
 });
 
+//
+// Load the initial api endpoint value from storage
+//
+chrome.storage.sync.get("work_available_endpoint", function(data) {
+  // Notify that we saved.
+  if (data['work_available_endpoint'] !== undefined) {
```
Comment (juliankrispel) [inline extension/js/main.js]: use . notation:

Instead of `data['bla']` data.bla

**Comment by juliankrispel** [comment]
@nxvl other than comments (the two comments apply to the whole file) tis good ;)

**Comment by juliankrispel** [comment]
:+1:

**Comment by jbarber** [comment]
@juliankrispel "javascript crime" LOL - love it :)

**Pull request merged**